nndownload/nndownload.py

```diff
@@ -1911,23 +1912,186 @@ def download_thumbnail(session: requests.Session, filename: AnyStr, template_par
     with open(filename, "w", encoding="utf-8") as file:
-        json.dump(get_comments_request.json(), file, indent=4, ensure_ascii=False, sort_keys=True)
+        json.dump(comments, file, indent=4, ensure_ascii=False, sort_keys=True)
 
     output("Finished downloading comments for {0}.\n".format(template_params["id"]), logging.INFO)
 
+def fetch_comments_modern(
+    session: requests.Session,
+    video_id: str,
+    api_server: str,
+    thread_key: str,
+    thread: dict,
+    language: str,
+    last_time: int,
+    limit: int,
+) -> dict:
+    """Fetch comments using modern Niconico API (requires login)."""
+    global_comments_data = {
+        "count": 0,
+        "id": thread["id"],
+    }
+    thread_data = {
+        "commentCount": 0,
+        "comments": [],
+        "fork": thread["fork"],
+        "id": thread["id"]
+    }
+    base_data = {
+        "threadKey": thread_key,
+        "params": {
+            "language": language,
+            "targets": [thread]
+        }
+    }
+    fetched_count = 0
+    
+    while fetched_count < limit:
+        try:
+            payload = {
+                **base_data,
+                "additionals": {
+                    "res_from": -1000,
+                    "when": last_time
+                }
+            }
+            
+            headers = {
+                "content-type": "text/plain;charset=UTF-8",
+                "x-client-os-type": "others",
+                "x-frontend-id": "6",
+                "x-frontend-version": "0"
+            }
+            
+            response = session.post(
+                f"{api_server}/v1/threads",
+                json=payload,
+                headers=headers
+            )
+            data = response.json()
+            
+            # Handle errors
+            if "meta" in data and "errorCode" in data["meta"]:
+                handle_api_error(data["meta"]["errorCode"], session, video_id)
+                continue
+
+            global_comment_count = data["data"]["globalComments"][0]["count"]
+            if global_comment_count and global_comment_count > global_comments_data["count"]:
+                # Get the highest global comment count as our estimated global threads total comment count
+                global_comments_data["count"] = global_comment_count
+
+            comment_count = data["data"]["threads"][0]["commentCount"]
+            if comment_count and fetched_count == 0:
+                # Use first fetch count to know the thread total comment count
+                thread_data["commentCount"] = comment_count 
+                
+            # Extract comments
+            thread_comments = data["data"]["threads"][0]["comments"]
+            # Not sure why this `no < 5` check was in the original comment-zouryou source code.
+            # I uncommented it out because some easy threads comments would not be added to the list.
+            if not thread_comments: # or thread_comments[0]["no"] < 5: 
```

Comment:
Just guessing, but maybe some system-level comments are numbered that low? I think it's fine to leave out for now